File: problems/problems_3686/solution.go
package problem3686

import (
	"encoding/json"
	"log"
	"strings"
)

const MOD = int(1e9 + 7)

func countStableSubsequences(nums []int) int {
	var dp0, dp1, dp00, dp01, dp10, dp11 int
	for _, num := range nums {
		if (num & 1) == 0 {
			dp0, dp00, dp10 = (dp0+1)%MOD, (dp00+dp0+dp10)%MOD, (dp10+dp1+dp01+dp11)%MOD
		} else {
			dp1, dp01, dp11 = (dp1+1)%MOD, (dp01+dp0+dp00+dp10)%MOD, (dp11+dp1+dp01)%MOD
		}
	}
	return (dp0 + dp1 + dp00 + dp01 + dp10 + dp11) % MOD
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return countStableSubsequences(nums)
}

File: problems/problems_3686/Solution.java
package problems.problems_3686;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    private static final int MOD = (int)1e9 + 7;
    public int countStableSubsequences(int[] nums) {
        long[][] f = new long[2][2];
        for (int x : nums) {
            x %= 2;
            f[x][1] = (f[x][1] + f[x][0]) % MOD;
            f[x][0] = (f[x][0] + f[x ^ 1][0] + f[x ^ 1][1] + 1) % MOD;
        }
        return (int) ((f[0][0] + f[0][1] + f[1][0] + f[1][1]) % MOD);
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] nums = jsonArrayToIntArray(inputJsonValues[0]);
        return JSON.toJSON(countStableSubsequences(nums));
    }
}
